packages/react-components/src/button/src/IconButton.tsx

```diff
@@ -1,75 +1,98 @@
 import "./IconButton.css";
 
 import { Box } from "../../box";
-import { Children, forwardRef } from "react";
+import { Children, ComponentProps, ElementType, ForwardedRef, ReactElement, SyntheticEvent } from "react";
 import { EmbeddedIcon } from "../../icons";
-import { any, bool, elementType, func, number, oneOf, oneOfType, string } from "prop-types";
-import { augmentElement, createEmbeddableAdapter, mergeProps, omitProps, slot } from "../../shared";
+import { augmentElement, createEmbeddableAdapter, forwardRef, mergeProps, omitProps, slot } from "../../shared";
 import { useButton } from "./useButton";
 import { useToolbarProps } from "../../toolbar";
 
-const propTypes = {
+export interface InnerIconButtonProps {
     /**
      * The icon button style to use.
      */
-    variant: oneOf(["solid", "outline", "ghost"]),
+    variant?: "solid" | "outline" | "ghost";
     /**
      * The icon button color accent.
      */
-    color: oneOf(["primary", "secondary", "danger", "inherit"]),
+    color?: "primary" | "secondary" | "danger" | "inherit";
     /**
      * The icon button shape.
      */
-    shape: oneOf(["rounded", "circular"]),
+    shape?: "rounded" | "circular";
     /**
      * Whether or not the icon button content should takes additional space.
      */
-    condensed: bool,
+    condensed?: boolean;
     /**
      * Whether or not the icon button should autoFocus on render.
      */
-    autoFocus: oneOfType([bool, number]),
+    autoFocus?: boolean | number
     /**
      * An icon button can show a loading indicator.
      */
-    loading: bool,
+    loading?: boolean;
     /**
      * An icon button can vary in size.
      */
-    size: oneOf(["2xs", "xs", "sm", "md"]),
+    size?: "2xs" | "xs" | "sm" | "md";
     /**
      * Whether or not the icon button is disabled.
      */
-    disabled: bool,
+    disabled?: boolean;
     /**
      * The icon button type.
      */
-    type: oneOf(["button", "submit", "reset"]),
+    type?: "button" | "submit" | "reset";
     /**
      * Called when the icon button is click.
      * @param {SyntheticEvent} event - React's original SyntheticEvent.
      * @returns {void}
      */
-    onClick: func,
+    onClick?(event: SyntheticEvent): void;
     /**
      * A label providing an accessible name to the icon button. See [WCAG](https://www.w3.org/TR/WCAG20-TECHS/ARIA14.html).
      */
-    "aria-label": string.isRequired,
+    "aria-label": string,
     /**
      * An HTML element type or a custom React element type to render as.
      */
-    as: oneOfType([string, elementType]),
+    as?: ElementType;
     /**
      * Default slot override.
      */
-    slot: string,
+    slot?: string,
+    /**
+     * Whether or not the button take up the width of its container.
+     */
+    fluid?: boolean;
+    /**
+     * @ignore
+     */
+    active?: boolean;
+    /**
+     * @ignore
+     */
+    focus?: boolean;
```

Comment:
Since `active`, `focus`, `hover` will never be documented, we could make a base interface for them?

Something like `InteractionStatesProps`

Reply:
Where do you think would be the best place to put that interface? 

`shared/interactiveStatesProps.ts`?
new folder `sharedTypes/interactiveStates.ts` 

other choice would be to make a ButtonInteractionStatesProps and put it in Button.tsx, with the other buttons referencing it

Reply:
shared/types/